src/devices/src/virtio/net/tap.rs

```diff
@@ -19,16 +19,26 @@ use utils::{ioctl_ioc_nr, ioctl_iow_nr};
 const IFACE_NAME_MAX_LEN: usize = 16;
 
 /// List of errors the tap implementation can throw.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum Error {
-    /// Unable to create tap interface.
-    CreateTap(IoError),
-    /// Invalid interface name.
-    InvalidIfname,
-    /// ioctl failed.
-    IoctlError(IoError),
-    /// Couldn't open /dev/net/tun.
+    /// Couldn't open /dev/net/tun
+    #[error("Couldn't open /dev/net/tun: {0}")]
```

Comment:
It generates the message based on the parameter passed in. This error can be used quite generically though, and could include issues with the `/sys/class` devices, in which case nothing would be wrong with the cloning device (`/dev/net/tun`). The issue here is that we are treating this as a catch all for all issues.

For a generic/catch-all error, it would be more appropriate to pass in parameters as:
```
"Unable to open network device [{}]: {}"
```

The first parameter being the network device target, in this case the tap name, followed by the error.

Reply:
Well, it is not like we open different files here. We always open `/dev/net/tun` and through it we configure net devices. So we can get this error regardless of the net device name.

Reply:
`/dev/net/tun` is the cloning device that tunnels are created from. However the individualistic properties of the tap device is associated with the `/sys/class/net/...` device path. I think this is emphasized by the limitation of needing to query this device to find out if it is a multi-queue device for example.

To be a bit more specific, it's the host device name that would be useful to provide here, see the parameter passed in to the network configuration [described in the docs](https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md?plain=1#L59).